components/org.wso2.carbon.identity.api.server.idp/org.wso2.carbon.identity.api.server.idp.v1/src/main/java/org/wso2/carbon/identity/api/server/idp/v1/core/ServerIdpManagementService.java

```diff
@@ -2521,19 +2521,17 @@ private JustInTimeProvisioning createJITResponse(IdentityProvider idp) {
         if (idp.getJustInTimeProvisioningConfig() != null) {
             jitConfig.setIsEnabled(idp.getJustInTimeProvisioningConfig().isProvisioningEnabled());
 
-            if (idp.getJustInTimeProvisioningConfig().isProvisioningEnabled()) {
-                boolean modifyUsername = idp.getJustInTimeProvisioningConfig().isModifyUserNameAllowed();
-                boolean passwordProvision = idp.getJustInTimeProvisioningConfig().isPasswordProvisioningEnabled();
-                boolean promptConsent = idp.getJustInTimeProvisioningConfig().isPromptConsent();
-                if (modifyUsername && passwordProvision && promptConsent) {
-                    jitConfig.setScheme(JustInTimeProvisioning.SchemeEnum.PROMPT_USERNAME_PASSWORD_CONSENT);
-                } else if (passwordProvision && promptConsent) {
-                    jitConfig.setScheme(JustInTimeProvisioning.SchemeEnum.PROMPT_PASSWORD_CONSENT);
-                } else if (promptConsent) {
-                    jitConfig.setScheme(JustInTimeProvisioning.SchemeEnum.PROMPT_CONSENT);
-                } else {
-                    jitConfig.setScheme(JustInTimeProvisioning.SchemeEnum.PROVISION_SILENTLY);
-                }
+            boolean modifyUsername = idp.getJustInTimeProvisioningConfig().isModifyUserNameAllowed();
+            boolean passwordProvision = idp.getJustInTimeProvisioningConfig().isPasswordProvisioningEnabled();
+            boolean promptConsent = idp.getJustInTimeProvisioningConfig().isPromptConsent();
```

Comment:
let's store the result of idp.getJustInTimeProvisioningConfig() in a local variable and reuse it

Reply:
also, are these null safe?

Reply:
Yeah, this is null same. If user does not specify any variable, pick the default values.